### PR TITLE
Best practices

### PR DESCRIPTION
**Pull request by jimflores5**
Here is the first draft for the 'best practices' chapter.

### CODE REVIEW

File: src/chapters/booster-rockets/bestpractices.rst [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+   classmates will give them a clear idea about your thought process.
+d. If you ran out of time, GO BACK AND FILL IN THE BLANKS LATER.  Practice
+   makes better.  If you neglect one set of skills, then the tasks that come
+   later and depend on those skills will be more difficult.
+
+.. admonition:: Take-home lesson:
+
+   Learning takes work, and you need the practice.  *Do your homework*.
+
+Sports Motivational Stuff
```
Comment (chrisbay) [inline src/chapters/booster-rockets/bestpractices.rst]: Per discussion: Look for non-semantic visual ways to differentiate between these quotes (list-table, list, etc).

File: src/chapters/booster-rockets/bestpractices.rst [hunk 2]
```diff
@@ -0,0 +1,232 @@
+Best Practices: Learning To Code
+=================================
+
+For years, the education community has debated the best approach for
+*"homework"*.  Should it be scored or not?  What is the proper amount?  Should
+it be even be assigned?  “Studies have shown…”
```
[the remark this answers is not in the document]
Comment (jimflores5) [reply src/chapters/booster-rockets/bestpractices.rst]: Rephrased the intro according to your suggestions.  Also dumbed down all the quotes.

File: src/chapters/booster-rockets/bestpractices.rst [hunk 3]
```diff
@@ -0,0 +1,232 @@
+Best Practices: Learning To Code
+=================================
+
+For years, the education community has debated the best approach for
+*"homework"*.  Should it be scored or not?  What is the proper amount?  Should
```
[the remark this answers is not in the document]
Comment (jimflores5) [reply src/chapters/booster-rockets/bestpractices.rst]: Fixed.

File: src/chapters/booster-rockets/bestpractices.rst [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+it be even be assigned?  “Studies have shown…”
+
+We cannot possibly restate all the conclusions here, nor will we present any
+part of the debates.  Instead, here are a few simple facts for LC101:
+
+#. Yes, there is homework, which consists of prep work, exercises, studios
```
[the remark this answers is not in the document]
Comment (jimflores5) [reply src/chapters/booster-rockets/bestpractices.rst]: Done.

File: src/chapters/booster-rockets/bestpractices.rst [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+   and assignments.
+#. Only the assignments earn a “score”.
```
[the remark this answers is not in the document]
Comment (jimflores5) [reply src/chapters/booster-rockets/bestpractices.rst]: Accepted and implemented rephrasing suggestion.

File: src/chapters/booster-rockets/bestpractices.rst [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+#. You do not have to do the ungraded homework, but you absolutely **SHOULD**.
```
[the remark this answers is not in the document]
Comment (jimflores5) [reply src/chapters/booster-rockets/bestpractices.rst]: My vote is still for double emphasis here.

File: src/chapters/booster-rockets/bestpractices.rst [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+
+Job Success
+-----------
+
+Imagine you land a sweet tech job, and on your first day your boss says to you,
+[some awful jumble of jargon gobbledygook]. If your entire prep for this job
+was reading without coding, you might understand each individual word, but
+actually accomplishing the task would be a disaster.
+
+Now imagine your prep involved using [relevant skills from gobbledygook], over
```
[the remark this answers is not in the document]
Comment (jimflores5) [reply src/chapters/booster-rockets/bestpractices.rst]: After trying to make up my own gobbledygook, I figured I'd wait for the experts to weigh in. ;-)

Suggestion accepted and pasted.

File: src/chapters/booster-rockets/bestpractices.rst [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+---------------------
+
+Does the following statement resonate with you?  If so, perfect.
+
+   "If I stumble, I WILL pick myself up, brush off the dust, and try again."
+
+All throughout LC101, you will be asked to code.  Each exercise, studio and
+assignment is designed to give you experience through practice.  You WILL make
+mistakes, and that is OK.  Often, our mistakes teach us more than getting the
+"correct" answer on our first try.
```
[the remark this answers is not in the document]
Comment (jimflores5) [reply src/chapters/booster-rockets/bestpractices.rst]: Fixed.

File: src/chapters/booster-rockets/bestpractices.rst [hunk 9]
```diff
@@ -0,0 +1,232 @@
...
+   a. The student did not understand how to solve the problem.
+   b. The student knew how to solve the problem and decided to skip the task.
+   c. The student tried to solve the problem, could not get the program to work,
+      so deleted the code.
+   d. The student ran out of time when trying to complete the prep work before
+      class.
+
+From a teacher’s perspective, ANY of these reasons could be valid, and we have
+no way of determining which is true.  This prevents us from knowing how to best
+help the student.  Where would be begin?
```
[the remark this answers is not in the document]
Comment (jimflores5) [reply src/chapters/booster-rockets/bestpractices.rst]: Fixed.

File: src/chapters/booster-rockets/bestpractices.rst [hunk 10]
```diff
@@ -0,0 +1,232 @@
...
+e. Your stickers demonstrate your commitment and might even inspire other
+   non-runners.  They will see how you started “just like them” and notice how
+   your effort spurred personal growth.
+
+Learning to code follows the same idea:
+
+a. Begin with “Hello world!”
+b. Then learn variables, strings and arrays.
+c. Then learn if/else statements and loops, followed by functions and modules.
+d. Then code your first half-marathon.
```
[the remark this answers is not in the document]
Comment (jimflores5) [reply src/chapters/booster-rockets/bestpractices.rst]: *"BOO!!!!!"*

File: src/chapters/booster-rockets/bestpractices.rst [hunk 11]
```diff
@@ -0,0 +1,232 @@
...
+*Whew!*  You made it to the bottom of the page.  Good job.  Here are some final bits of advice:
+   #. DO try every exercise, studio and practice problem.
+   #. Repeated practice helps master the basic syntax quirks for a given
+      programming language.
+   #. DO experiment.  Once your code correctly solves the given task, feel
+      free to tweak it.  Great fun can be had if you ask, “What if I try ____,”
+      and then go and do just that.  For example, if a problem asks you to
+      sort a list alphabetically, can you order it from z to a instead?
+   #. ASK FOR HELP when you get stuck.  We’ve all been there, and there is
+      no shame in seeking advice.  Use your instructors, TA’s, classmates and
```
[the remark this answers is not in the document]
Comment (jimflores5) [reply src/chapters/booster-rockets/bestpractices.rst]: I go back and forth myself.  TAs implemented.

File: src/chapters/booster-rockets/bestpractices.rst (hunk 1 above)
Comment (jimflores5) [reply src/chapters/booster-rockets/bestpractices.rst]: Placed quotes into a table with alternate shaded rows.

**Pull request merged**